src/include/nanobench.h

```diff
@@ -2032,14 +2033,15 @@ void gatherStabilityInformation(std::vector<std::string>& warnings, std::vector<
     if (nprocs <= 0) {
         warnings.emplace_back("couldn't figure out number of processors - no governor, turbo check possible");
     } else {
+        auto fail = false;
 
         // check frequency scaling
         for (long id = 0; id < nprocs; ++id) {
             auto idStr = detail::fmt::to_s(static_cast<uint64_t>(id));
             auto sysCpu = "/sys/devices/system/cpu/cpu" + idStr;
-            auto minFreq = parseFile<int64_t>(sysCpu + "/cpufreq/scaling_min_freq");
-            auto maxFreq = parseFile<int64_t>(sysCpu + "/cpufreq/scaling_max_freq");
-            if (minFreq != maxFreq) {
+            auto minFreq = parseFile<int64_t>(sysCpu + "/cpufreq/scaling_min_freq", &fail);
+            auto maxFreq = parseFile<int64_t>(sysCpu + "/cpufreq/scaling_max_freq", &fail);
```

Comment:
I don't think the &fail handling here is necessary for scaling_min_freq and scaling_max_freq. If these files are not there, minFreq will be equal to maxFreq (both zero) and no warning is added. I'd remove it here, but keep it for the others (scaling governor and turbo). Also because the second call overwrites the state of &fail anyways

Reply:
Good point, pushed an update. Let me know what you think!